applications/RANSApplication/custom_strategies/algebraic_flux_corrected_scalar_steady_scheme.h

```diff
@@ -0,0 +1,567 @@
+                                p_plus[i] += std::max(0.0, f_ij);
+                                p_minus[i] -= std::max(0.0, -f_ij);
+                            }
+
+                            if (equation_ids[i] < equation_ids[j])
+                            {
+                                q_plus[i] += std::max(0.0, -f_ij);
+                                q_minus[i] -= std::max(0.0, f_ij);
+                                q_plus[j] += std::max(0.0, f_ij);
+                                q_minus[j] -= std::max(0.0, -f_ij);
+                            }
+                        }
+                    }
+                }
+
+                for (int i = 0; i < size; ++i)
+                {
+                    ModelPart::NodeType& r_node = r_geometry[i];
+                    r_node.SetLock();
+                    r_node.FastGetSolutionStepValue(AFC_POSITIVE_ANTI_DIFFUSIVE_FLUX) +=
+                        p_plus[i];
+                    r_node.FastGetSolutionStepValue(AFC_POSITIVE_ANTI_DIFFUSIVE_FLUX_LIMIT) +=
+                        q_plus[i];
+                    r_node.FastGetSolutionStepValue(AFC_NEGATIVE_ANTI_DIFFUSIVE_FLUX) +=
+                        p_minus[i];
+                    r_node.FastGetSolutionStepValue(AFC_NEGATIVE_ANTI_DIFFUSIVE_FLUX_LIMIT) +=
+                        q_minus[i];
+                    r_node.UnSetLock();
+                }
+            }
+        }
+
+        if (mrPeriodicIdVar != Variable<int>::StaticObject())
+        {
+            const int number_of_conditions = rModelPart.NumberOfConditions();
+#pragma omp parallel for
+            for (int i_cond = 0; i_cond < number_of_conditions; ++i_cond)
+            {
+                ModelPart::ConditionType& r_condition =
+                    *(rModelPart.ConditionsBegin() + i_cond);
+                if (r_condition.Is(PERIODIC))
+                {
+                    ModelPart::NodeType& r_node_0 = r_condition.GetGeometry()[0];
+                    ModelPart::NodeType& r_node_1 = r_condition.GetGeometry()[1];
+
+                    double p_plus = r_node_0.FastGetSolutionStepValue(AFC_POSITIVE_ANTI_DIFFUSIVE_FLUX);
+                    double q_plus = r_node_0.FastGetSolutionStepValue(
+                        AFC_POSITIVE_ANTI_DIFFUSIVE_FLUX_LIMIT);
+                    double p_minus =
+                        r_node_0.FastGetSolutionStepValue(AFC_NEGATIVE_ANTI_DIFFUSIVE_FLUX);
+                    double q_minus = r_node_0.FastGetSolutionStepValue(
+                        AFC_NEGATIVE_ANTI_DIFFUSIVE_FLUX_LIMIT);
+
+                    p_plus += r_node_1.FastGetSolutionStepValue(AFC_POSITIVE_ANTI_DIFFUSIVE_FLUX);
+                    q_plus += r_node_1.FastGetSolutionStepValue(AFC_POSITIVE_ANTI_DIFFUSIVE_FLUX_LIMIT);
+                    p_minus += r_node_1.FastGetSolutionStepValue(AFC_NEGATIVE_ANTI_DIFFUSIVE_FLUX);
+                    q_minus += r_node_1.FastGetSolutionStepValue(AFC_NEGATIVE_ANTI_DIFFUSIVE_FLUX_LIMIT);
```

Comment:
Couldn't these variable be non-historical?

Reply:
I added them since it is faster. They can be non-historical also.

Reply:
done :+1: